encrypt/encrypt.go

```diff
@@ -75,14 +77,38 @@ func Encrypt(args []string) error {
 		return err
 	}
 
-	// no key provided, check for one in the session file
-	if len(publicKeyFileList) == 0 {
+	var pubKey string
+
+	if publicKeyFileList != nil && *target != "" {
+		return errors.New("only one of -key or -target can be used")
+	}
 
-		sesKey, err := helpers.GetPublicKey()
+	if *target != "" {
+		// fetch info endpoint values
+		info, err := login.GetAuthInfo(*target)
+		if err != nil {
+			return err
+		}
+		// create pub file
+		pubKeyFile, err := helpers.CreatePubFile(info.PublicKey, "crypt4gh_key.pub")
+		if err != nil {
+			return err
+		}
+		log.Println("fetching public key")
+		// no key provided, no key in session file, target provided
+		publicKeyFileList = append(publicKeyFileList, pubKeyFile)
+	}
+	// no key provided, no key in session file, no target provided
+	if publicKeyFileList == nil && *target == "" {
+		// check for public key in .sda-cli-session file from login
+		pubKey, err = helpers.GetPublicKeyFromSession()
```

Comment:
```suggestion
		pubKey, err := helpers.GetPublicKeyFromSession()
```

Do full declaration here